module_2/gift_certificates_system/service/src/main/java/com/epam/esm/service/impl/GiftCertificateServiceImpl.java

```
@@ -0,0 +1,254 @@
+    public GiftCertificate create(GiftCertificate entity) {
+        String name = entity.getName();
+        Optional<GiftCertificate> optionalGiftCertificate = giftCertificateDao.findByName(name);
+
+        if(optionalGiftCertificate.isPresent()) {
+            log.debug("Gift certificate with name '{}' already exists in the database.", entity.getName());
+            throw new EntityExistsException("Gift certificate with name '" + entity.getName() + "' already exists in the database.");
+        }
+
+        LocalDateTime currentLocalDateTime = LocalDateTime.now();
+        entity.setCreateDate(currentLocalDateTime);
+        entity.setLastUpdateDate(currentLocalDateTime);
+
+        GiftCertificate createdGiftCertificate = giftCertificateDao.create(entity);
+        createRelationsBetweenGiftCertificateAndTags(createdGiftCertificate);
+
+        return createdGiftCertificate;
+    }
+
+    @Override
+    public List<GiftCertificate> findAll() {
+        List<GiftCertificate> giftCertificates = giftCertificateDao.findAll();
+        giftCertificates = getGiftCertificatesWithTags(giftCertificates);
+
+        return giftCertificates;
+    }
+
+    @Override
+    public List<GiftCertificate> findAllAndSort(ColumnName columnName, SortOperator sortOperator) {
+        List<GiftCertificate> giftCertificates = findAll();
+
+        switch (columnName) {
+            case NAME:
+                giftCertificates.sort(Comparator.comparing(GiftCertificate::getName));
+                break;
+            case CREATE_DATE:
+                giftCertificates.sort(Comparator.comparing(GiftCertificate::getCreateDate));
+                break;
+            case LAST_UPDATE_DATE:
+                giftCertificates.sort(Comparator.comparing(GiftCertificate::getLastUpdateDate));
+                break;
+            case WITHOUT_SORTING:
+                break;
+            default:
+                throw new EnumConstantNotPresentException(ColumnName.class, columnName.name());
```

why is this exception is not handled?

---

Thanks, fixed.